src/confluent_kafka/src/Admin.c

```diff
@@ -976,6 +976,537 @@ static PyObject *Admin_alter_configs (Handle *self, PyObject *args,
+                        resource_pattern_type, principal, host, \
+                        operation, permission_type, errstr, errstr_size);
+        }
+        if (resname) free(resname);
+        if (principal) free(principal);
+        if (host) free(host);
+        return ret;
+}
+
+/**
+ * @brief Convert C AclBinding to py
+ */
+static PyObject *
+Admin_c_AclBinding_to_py (const rd_kafka_AclBinding_t *c_acl_binding) {
+        
+        PyObject *args, *kwargs, *AclBinding_type, *acl_binding;
+
+        AclBinding_type = cfl_PyObject_lookup("confluent_kafka.admin",
+                                        "AclBinding");
+        
+        kwargs = PyDict_New();
+
+        cfl_PyDict_SetInt(kwargs, "restype",
+                                     rd_kafka_AclBinding_restype(c_acl_binding));
+        cfl_PyDict_SetString(kwargs, "name",
+                                     rd_kafka_AclBinding_name(c_acl_binding));
+        cfl_PyDict_SetInt(kwargs, "resource_pattern_type",
+                                rd_kafka_AclBinding_resource_pattern_type(c_acl_binding));
+        cfl_PyDict_SetString(kwargs, "principal",
+                                     rd_kafka_AclBinding_principal(c_acl_binding));
+        cfl_PyDict_SetString(kwargs, "host",
+                                     rd_kafka_AclBinding_host(c_acl_binding));
+        cfl_PyDict_SetInt(kwargs, "operation",
+                                     rd_kafka_AclBinding_operation(c_acl_binding));
+        cfl_PyDict_SetInt(kwargs, "permission_type",
+                                     rd_kafka_AclBinding_permission_type(c_acl_binding));
+
+        args = PyTuple_New(0);
+        acl_binding = PyObject_Call(AclBinding_type, args, kwargs);
+
+        Py_DECREF(args);
+        Py_DECREF(kwargs);
+        Py_DECREF(AclBinding_type);
+        return acl_binding;
+}
+
+/**
+ * @brief Convert C AclBinding array to py list.
+ */
+static PyObject *
+Admin_c_AclBindings_to_py (const rd_kafka_AclBinding_t **c_acls,
+                                          size_t c_acls_cnt) {
+        size_t i;
+        PyObject *result;
+
+        result = PyList_New(c_acls_cnt);
+
+        for (i = 0 ; i < c_acls_cnt ; i++) {
+                PyList_SET_ITEM(result, i,
+                        Admin_c_AclBinding_to_py(c_acls[i]));
+        }
+
+        return result;
+}
+
+
+/**
+ * @brief Convert C acl_result_t array to py list.
+ */
+static PyObject *
+Admin_c_acl_result_to_py (const rd_kafka_acl_result_t **c_result,
+                            size_t cnt) {
+        PyObject *result;
+        size_t ti;
+
+        result = PyList_New(cnt);
+
+        for (ti = 0 ; ti < cnt ; ti++) {
+                PyObject *error;
+                const rd_kafka_error_t *c_error = rd_kafka_acl_result_error(c_result[ti]);
+
+                error = KafkaError_new_or_None(
+                        rd_kafka_error_code(c_error),
+                        rd_kafka_error_name(c_error));
```

Comment:
this should be error_string - i.e., the human readable error description.  _name is just the enum name.